### PR TITLE
Enable context isolation

### PR DESCRIPTION
**Pull request by nh758**
Adds #243

### CODE REVIEW

File: main/electron.ts [hunk 1]
```diff
@@ -32,7 +32,13 @@ export function createWindow(): void {
     height: 970,
     show: false,
     backgroundColor: '#30404d',
-    webPreferences: { nodeIntegration: true, webSecurity: true, enableRemoteModule: false },
+    webPreferences: {
+      nodeIntegration: false,
+      webSecurity: true,
+      enableRemoteModule: true,
```
Comment (irahopkinson) [inline main/electron.ts:38]: It took a lot of work to get the remote module disabled. We don't want it enabled again.

File: src/App/store/AppState.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { ipcRenderer } from 'electron';
+// import { ipcRenderer } from 'electron';
```
Comment (irahopkinson) [inline src/App/store/AppState.ts:1]: Remove this line rather than commenting it.

File: main/preload.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   * without security problems.
+   *
+   * The functions below can accessed using `window.api.fnName`
+   */
+
+  getBKProject: (rootDirectories: RootDirectories): void => {
+    ipcRenderer.send('did-start-getbkproject', rootDirectories);
+  },
+
+  onBKProject: (callback: Function): void => {
```
Comment (irahopkinson) [inline main/preload.ts:21]: I think the linter might complain about the use of `Function` as a type. Either way, could we use a more specific type, i.e. `(project: BKProject) => void`. Similarly below.

File: main/preload.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    ipcRenderer.on('did-finish-getbkproject', (_event: Event, projects: BKProject) => {
```
Comment (irahopkinson) [inline main/preload.ts:22]: Its just one `project` (rather than plural `projects`).

File: src/App/store/AppState.ts [hunk 5]
```diff
@@ -226,7 +226,7 @@ export class Project implements BKProject {
 class ProjectList {
   constructor() {
     makeObservable(this);
-    ipcRenderer.on('did-finish-getbkproject', (_event: Event, projects: Project[]) => {
+    window.api.onBKProject((projects: Project[]) => {
```
Comment (irahopkinson) [inline src/App/store/AppState.ts:229]: `projects` should also be singular here also (I know it was already like that but good too fix since we noticed).

**Review by irahopkinson — CHANGES_REQUESTED** [review]
This is looking good. A couple of things to fix up.

File: main/preload.ts (hunk 4 above)
Comment (nh758) [reply main/preload.ts]: It should actually returns an array. The type here was wrong.

**Review by irahopkinson — APPROVED** [review]
👍 LGTM. Thanks for this.

**Pull request merged**